search/includes/classes/class-health.php

```diff
@@ -819,9 +815,18 @@ public function get_index_settings_diff_for_indexable( \ElasticPress\Indexable $
 			$diff = self::get_index_settings_diff( $actual_settings_to_check, $desired_settings_to_check );
 		}
 
+		$result = [];
+		if ( ! empty($diff) ) {
+			$result = array(
+				'index_version' => $options['index_version'] ?? 1,
+				'index_name' => $indexable->get_index_name(),
+				'diff' => $diff,
+			);
+		}
+
 		$this->search->versioning->reset_current_version_number( $indexable );
```

Comment:
Now this happens only after we have done `$indexable->get_index_name()` few lines above, so all is good with the world again.

Reply:
well done.